openedx/edx-ace
# feat: [FC-0047] add functionality to send push notifications

**Description:**
This PR is addressed at adding a new channel that can send push notifications to the OeX mobile app.
A more detailed description can be found in the relevant ADR here - https://github.com/openedx/edx-ace/pull/277

**JIRA:** FC-0047

**Dependencies:** https://github.com/openedx/edx-platform/pull/34971

**Testing instructions:**

1. Install edx-ace version 1.9.0.
2. Checkout the edx-platform from the [implement-push-notifications-chanel](https://github.com/raccoongang/edx-platform/tree/NiedielnitsevIvan/FC-0047/feature/implement-push-notifications-chanel) branch and rebuild.
3. Add FIREBASE_CREDENTIALS or FIREBASE_CREDENTIALS_PATH to the platform settings.
4. Add a new `push_notification` channel to ACE_ENABLED_CHANEL.
5. Log in to OeX.
6. Install the OeX mobile application on your device.
7. Add your device token to FCM Devices with a link to your user from the admin panel.
8. Enroll/unenroll the user(from step 7) in the “Instructor” tab to trigger the push notification event.
9. Make sure you receive push notifications on your device.

**Reviewers:**
- [ ] @e0d 
- [ ] @OmarIthawi 
- [ ] @bmtcril 

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** 
This PR and the PR that adds the settings for the firebase-admin and django-push-notification package in edx-platform should be merged at the same time to avoid errors.
